apps/examples/log_dump/log_dump_main.c

```diff
@@ -42,37 +52,47 @@ int main(int argc, FAR char *argv[])
 int log_dump_main(int argc, char *argv[])
 #endif
 {
+	if (argc != 2) {
+		show_usage();
+		goto errout;
+	}
+
 	int ret = 1;
 	char buf[READ_BUFFER_SIZE];
 	int fd = OPEN_LOGDUMP();
 
 	if (fd < 0) {
-		printf("Please ensure that procfs is mounted\n");
-		return -1;
+		printf("log_dump_main : Please ensure that procfs is mounted\n");
+		goto errout;
 	}
 
 	/* Log dump starts automatically from boot up.
 	 * To test start, intentionally add to stop. */
 	if (STOP_LOGDUMP_SAVE(fd) < 0) {
-		printf("Failed to stop log dump, errno %d\n", get_errno());
+		printf("log_dump_main : Failed to stop log dump, errno %d\n", get_errno());
+		goto errout;
+	}
+
+	if (GET_LOGDUMP_FROM_LAST(fd, strtol(argv[1], NULL, 10))) {
```

Comment:
No need to call `READ_LOGDUMP` for all?
If it is, why do we keep it on the header?
Don't we need to split the function for all and partial?

Reply:
```GET_LOGDUMP_FROM_LAST(fd, N)``` acts like ```file.seek(fd, offset)``` . When anyone who wants to read the log dump data opens the ```"proc/logsave"``` file, they can use ```GET_LOGDUMP_FROM_LAST``` similar to file seek to move the read pointer. We will still need ```READ_LOGDUMP``` to read the file after doing file seek. May be ```GET_LOGDUMP_FROM_LAST``` is not the right name. I will change it to ```LOGDUMP_SEEK```

Reply:
Because of dynamic memory management, if we read some nodes using this and then read again after sometime, we can't guarantee previous read point.
I mean 
1. Let's guess there are 5 nodes.
2. Read last 2 nodes. (index 3,4)
3. After sometime, read 2 nodes (index 1,2)

In this case, 1,2,3,4 would be not continuous logs almost.
So, I think we can support just read from last always, without seek functionality.
How do you think?

Reply:
> When anyone who wants to read the log dump data opens the "proc/logsave" file, they can use GET_LOGDUMP_FROM_LAST similar to file seek to move the read pointer. We will still need READ_LOGDUMP to read the file after doing file seek.

The sample application should show all of supported functionalities but it has `read_from_last` only, there is no `read_all`. This is why I left a comment.

Reply:
> Because of dynamic memory management, if we read some nodes using this and then read again after sometime, we can't guarantee previous read point. I mean
> 
> 1. Let's guess there are 5 nodes.
> 2. Read last 2 nodes. (index 3,4)
> 3. After sometime, read 2 nodes (index 1,2)
> 
> In this case, 1,2,3,4 would be not continuous logs almost. So, I think we can support just read from last always, without seek functionality. How do you think?

Currently, we always reset the read pointer. So when someone closes the log dump fd and re-opens it, it doesn't store the old read pointer.
```0 -> 1 -> 2 -> 3-> 4-> 5```
first time when user calls log_dump 2, then 4 and 5 are fetched.
when user reopens the file and calls log_dump 2, then again 4 and 5 are fetched.

This is how the current implementation works. I thought this was the required functionality as even earlier during log dump we were always fetching all nodes from 0 to 5, irrespective of how much was read before. Please let me know if I missed something regarding the requirement.

Reply:
> Currently, we always reset the read pointer. So when someone closes the log dump fd and re-opens it, it doesn't store the old read pointer.
0 -> 1 -> 2 -> 3-> 4-> 5
first time when user calls log_dump 2, then 4 and 5 are fetched.
when user reopens the file and calls log_dump 2, then again 4 and 5 are fetched.
>
> This is how the current implementation works. I thought this was the required functionality as even earlier during log dump we were always fetching all nodes from 0 to 5, irrespective of how much was read before. Please let me know if I missed something regarding the requirement.

This looks ok. No need to read middle node. We always read last # nodes (I mean seek functionality is unnecessary).

Reply:
Okay